### ISSUE TITLE
PTH false positive with Django FieldFile

### ISSUE THREAD

**Issue by gustavi** [astral-sh/ruff#4981]
Example of false positive for `PTH110` and `PTH116`:

```python
import os
from django.db import models
from django.test import TestCase


class Document(models.Model):
    file = models.FileField()


class MyTest(TestCase):
    def test_foo(self):
        file_path = Document.objects.last().file.path
        self.assertTrue(os.path.exists(file_path))
        self.assertEqual(os.stat(file_path).st_mode & 0o777, 0o660)

```

Output :

```
test.py:13:25: PTH110 `os.path.exists()` should be replaced by `Path.exists()`
test.py:14:26: PTH116 `os.stat()` should be replaced by `Path.stat()`, `Path.owner()`, or `Path.group()
```

`Document.objects.last().file.path` is `str`.

Tested on `ruff 0.0.272`.

Django code : https://github.com/django/django/blob/4.2/django/db/models/fields/files.py#L60

_Note: sample code can not be launched, I can provide a full working example with Django if you really need._

**Comment by dhruvmanila** [comment]
I think the diagnostics are correct. What it's suggesting is the following:

```python
file_path = "foo/bar.py"

os.path.exists(file_path)

# Instead use
Path(file_path).exists()
```

Do you think it could be useful to include the variable name in the message like so:
```
test.py:13:25: PTH110 `os.path.exists(file_path)` should be replaced by `Path(file_path).exists()`
test.py:14:26: PTH116 `os.stat(file_path)` should be replaced by `Path(file_path).stat()`, `Path(file_path).owner()`, or `Path(file_path).group()
```

**Comment by dhruvmanila** [comment]
No problem at all :)
